guides/plugins/apps/tax-provider.md

```diff
@@ -0,0 +1,140 @@
+# Tax provider
+
+Tax calculations differ from country to country. Especially in the US, the sales tax calculation can be tedious, as the laws and regulation differ from state to state or even county or certain city taxes can apply. Therefore, most shops use a third-party service (so called tax-providers) to calculate sales taxes for them.
+
+Starting with version 6.5.0.0, Shopware allows apps to integrate custom tax calculation, which could include an automatic tax calculation with a tax provider. An app has to provide an endpoint, which is called during checkout and provide new tax rates. The requests and responses of all of your endpoints will be signed and feature JSON content.
+
+## Prerequisites
+
+You should be familiar with the concept of Apps and their registration.
+
+{% page-ref page="app-base-guide.md" %}
+
+To reproduce this example, you should also be aware of how to set up an app on your development platform.
+
+{% page-ref page="local-development/" %}
+
+## Manifest configuration
+
+To signalize Shopware, that your app uses a custom tax calculation, you have to provide one or more `tax-provider` properties inside a `tax` parent property in your app's `manifest.xml`.
+
+Below, you can see an example definition of a working tax-provider.
+
+{% code title="manifest.xml" %}
+
+```markup
+<?xml version="1.0" encoding="UTF-8"?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+        <!-- The name of the app should not change. Otherwise all payment methods are created as duplicates. -->
+        <name>PaymentApp</name>
+        <!-- ... -->
+    </meta>
+    
+    <tax>
+        <tax-provider>
+            <identifier>myCustomTaxProvider</identifier>                        <!-- Unique identifier of the tax provider -->
+            <name>My custom tax provider</name>                                 <!-- Display name of the tax provider -->    
+            <priority>1</priority>                                              <!-- Priority of the tax provider - can be changed in the administration as well -->
+            <processUrl>https://tax-provider.app/provide-taxes</processUrl>     <!-- Url of your implementation - is called during checkout to provide taxes -->
+        </tax-provider>
+    </tax>
+</manifest>
+```
+
+{% endcode %}
+
+After successful installation of your app, the tax provider will already be used during checkout to provide taxes.
```

Comment:
> you have to provide one or more `tax-provider` properties

So will all be active at once? Can I also make them sort of optionally enabled? I mean when I try to solve my tax problem and I am in the Shopware cloud I have to try out different tax provider service I will install some apps. How do I know which tax provider calculated it and in which order they work together? Is this something an app can influence or is it done in the admin?

Edit: I've later read about it in the plugin explanation. They need to share maybe a concept page, that also "end users" can understand?

Reply:
Good catch, it is not possible to create inactive tax providers, which is an issue, I guess :laughing: 
I will put this to a discussion and we will probably default to inactive providers, which have to be activated in the administration first.

For the order of tax providers: This can be changed by the shopowner via the administration, therefore, a tax provider extension can not rely on having a certain priority during checkout.

There will be an end user documentation, I will redirect your concerns to our end user technical writers.